app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,29 @@
+class CommentsController < ApplicationController
+  def new
+    @comment = Comment.new
+    @user = current_user
+  end
+
+  def create
+    @post = Post.find(params[:post_id] || params[:id])
+    @comment = Comment.new(text: params[:comment][:text], post_id: @post.id, author_id: current_user.id)
+    @comment.author_id = current_user.id
+
+    respond_to do |format|
+      if @comment.save
+        format.html do
+          redirect_to user_post_url(current_user, @post), notice: 'Comment was successfully created.'
+        end
+      else
+        format.html { render :new, status: :unprocessable_entity }
+      end
+    end
+
+    # rubocop:disable Lint/NestedMethodDefinition
+    def comment_params
+      params.require(:comment).permit(:text)
+    end
+
+    # rubocop:enable Lint/NestedMethodDefinition
```

Comment:
- [ ] This strong parameter method is nested in the `create` action, but that should not be so. Please remove this method from the action and make it private, as you have done in with `post_params` in the posts controller.